json/marshal.go

```diff
@@ -0,0 +1,193 @@
+package marshal
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+const FORCE_SEND_FIELD_NAME = "ForceSendFields"
+
+// Marshal returns a JSON encoding of the given object. Included fields:
+// - non-empty value
+// - a basic type whose field's name is present in forceSendFields
+// Our templates always populate either omitempty or the '-' tag.
+// For simplicity, we assume one of those cases.
+func Marshal(object interface{}) ([]byte, error) {
+
+	dataMap, err := structAsMap(object)
+	if err != nil {
+		return nil, err
+	}
+
+	return json.Marshal(dataMap)
+}
+
+// Converts the object to a map
+func structAsMap(object interface{}) (map[string]interface{}, error) {
+	result := make(map[string]interface{})
+
+	// If the object is nil or a pointer to nil, don't do anything
+	if object == nil || (reflect.ValueOf(object).Kind() == reflect.Ptr && reflect.ValueOf(object).IsNil()) {
+		return result, nil
+	}
+
+	forceSendFields, err := getForceSendFields(object)
+
+	if err != nil {
+		return nil, err
+	}
+
+	includeFields := make(map[string]bool)
+	for _, field := range forceSendFields {
+		includeFields[field] = true
```

Comment:
The map only contains `forceSendFields`, thus the `getForceSendFields` function can return a map.